test_util/aws.py

```diff
@@ -230,7 +230,11 @@ def create(cls, stack_name, boto_wrapper, template_url,
             'PrivateAgentInstanceType': private_agent_type,
             'PrivateSubnet': private_subnet}
         stack = boto_wrapper.create_stack(stack_name, template_url, parameters)
-        return cls(stack.stack.stack_name, boto_wrapper), SSH_INFO['coreos']
+        if template_url.split('/')[-1].startswith('coreos'):
+            os_name = 'coreos'
+        else:
+            os_name = 'centos'
```

Comment:
If we add a template that uses a different OS that this code doesn't expect, it'd be easier to debug an exception here rather than an SSH error later. It looks like the advanced templates have their own set of aliases for AMIs, so let's add a `CF_OS_SSH_INFO` dict alongside the others with the keys `'coreos'` and `'el7'`. Then this can return `CF_OS_SSH_INFO[template_url.split('/')[-1].split('-')[0]]`. Those `split()`s and indexes assume a lot about the URL structure that isn't asserted elsewhere in this method, so it might be useful to handle `(KeyError, IndexError)` there by raising an exception that mentions the template URL.